### PR TITLE
GPT pre-train script update for fp8 training

### PR DESCRIPTION
**Pull request by erhoo82**
- Set the fp8-associated knobs for easy running of fp8 jobs
- Update README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1366,7 +1366,7 @@ creating the job (number of replicas).
 Transformer Engine (TE) is a library for accelerating Transformer-based models on **NVIDIA Hopper GPUs**. It enables using 8-bit floating point (FP8) precision to provide better performance with lower memory utilization in both training and inference. NVIDIA open-sourced TE on [github](https://github.com/NVIDIA/TransformerEngine).
 
 In NeMo Framework, you can now use `fp8` to pre-train GPT models. For example, if you want to turn on `fp8` to pre-train a 
-GPT3 5B model, you can modify `gpt3/5b` training config inside `conf/training/gpt3/5b.yaml` file as following.
+GPT3 5B model, you can modify `gpt3/5b` training config inside `conf/training/gpt3/5b.yaml` file as following. We have already set the knobs associated with fp8 training accordingly, so all you need to change from the baseline yaml script for pre-training is setting `fp8` to `True`. In case of bf16 training, these knobs are ignored.
```
Comment (ericharper) [inline README.md:1369]: Let's mention `transformer_engine=True` and `use_fp8=True` is what's needed

File: launcher_scripts/conf/training/gpt3/126m.yaml [hunk 2]
```diff
@@ -109,14 +109,16 @@ model:
   grad_allreduce_chunk_size_mb: 125
 
   ## Transformer Engine
-  transformer_engine: False
+  # To use fp8, change only `fp8` to `True`.
+  # The rest of fp8 knobs are set for the fp8 training mode, which are ignored in non-fp8 training
+  transformer_engine: True
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/126m.yaml:114]: Can we revert this to `False` as `True` is currently breaking CI until we fix TE + PEFT

File: launcher_scripts/conf/training/gpt3/1b_improved.yaml [hunk 3]
```diff
@@ -126,14 +126,16 @@ model:
   gc_interval: 100 # Interval of the host memory garbage collection
 
   ## Transformer Engine
-  transformer_engine: False
+  # To use fp8, change only `fp8` to `True`.
+  # The rest of fp8 knobs are set for the fp8 training mode, which are ignored in non-fp8 training
+  transformer_engine: True
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/1b_improved.yaml:131]: all improved configs should have `transformer_engine=False` as they don't yet work with TE due to lack of rope and swiglu support

File: launcher_scripts/conf/training/gpt3/400m_improved.yaml [hunk 4]
```diff
@@ -126,14 +126,16 @@ model:
   gc_interval: 100 # Interval of the host memory garbage collection
 
   ## Transformer Engine
-  transformer_engine: False
+  # To use fp8, change only `fp8` to `True`.
+  # The rest of fp8 knobs are set for the fp8 training mode, which are ignored in non-fp8 training
+  transformer_engine: True
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/400m_improved.yaml:131]: all improved configs should have transformer_engine=False as they don't yet work with TE due to lack of rope and swiglu support

File: launcher_scripts/conf/training/gpt3/40b_improved.yaml [hunk 5]
```diff
@@ -128,14 +128,16 @@ model:
   gc_interval: 100 # Interval of the host memory garbage collection
   
   ## Transformer Engine
-  transformer_engine: False
+  # To use fp8, change only `fp8` to `True`.
+  # The rest of fp8 knobs are set for the fp8 training mode, which are ignored in non-fp8 training
+  transformer_engine: True
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/40b_improved.yaml:133]: all improved configs should have transformer_engine=False as they don't yet work with TE due to lack of rope and swiglu support

File: launcher_scripts/conf/training/gpt3/7b_improved.yaml [hunk 6]
```diff
@@ -126,14 +126,16 @@ model:
   gc_interval: 100 # Interval of the host memory garbage collection
 
   ## Transformer Engine
-  transformer_engine: False
+  # To use fp8, change only `fp8` to `True`.
+  # The rest of fp8 knobs are set for the fp8 training mode, which are ignored in non-fp8 training
+  transformer_engine: True
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/7b_improved.yaml:131]: all improved configs should have transformer_engine=False as they don't yet work with TE due to lack of rope and swiglu support

update comment to include `# To use fp8, `transformer_engine=True` and `fp8`=True

File: launcher_scripts/conf/training/gpt3/5b.yaml [hunk 7]
```diff
@@ -108,14 +108,16 @@ model:
   grad_allreduce_chunk_size_mb: 125
 
   ## Transformer Engine
+  # To use fp8, change only `fp8` to `True`.
+  # The rest of fp8 knobs are set for the fp8 training mode, which are ignored in non-fp8 training
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/5b.yaml:112]: update comment to include # To use fp8, transformer_engine=True and fp8=True

File: launcher_scripts/conf/training/gpt3/40b.yaml [hunk 8]
```diff
@@ -108,14 +108,16 @@ model:
   grad_allreduce_chunk_size_mb: 125
 
   ## Transformer Engine
+  # To use fp8, change only `fp8` to `True`.
+  # The rest of fp8 knobs are set for the fp8 training mode, which are ignored in non-fp8 training
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/40b.yaml:112]: update comment to include # To use fp8, transformer_engine=True and fp8=True

Comment (erhoo82) [reply launcher_scripts/conf/training/gpt3/40b.yaml]: Done

File: launcher_scripts/conf/training/gpt3/5b.yaml (hunk 7 above)
Comment (erhoo82) [reply launcher_scripts/conf/training/gpt3/5b.yaml]: Done

File: launcher_scripts/conf/training/gpt3/7b_improved.yaml (hunk 6 above)
Comment (erhoo82) [reply launcher_scripts/conf/training/gpt3/7b_improved.yaml]: Done

File: launcher_scripts/conf/training/gpt3/40b_improved.yaml (hunk 5 above)
Comment (erhoo82) [reply launcher_scripts/conf/training/gpt3/40b_improved.yaml]: Done

File: launcher_scripts/conf/training/gpt3/400m_improved.yaml (hunk 4 above)
Comment (erhoo82) [reply launcher_scripts/conf/training/gpt3/400m_improved.yaml]: Done

File: launcher_scripts/conf/training/gpt3/1b_improved.yaml (hunk 3 above)
Comment (erhoo82) [reply launcher_scripts/conf/training/gpt3/1b_improved.yaml]: Done

File: launcher_scripts/conf/training/gpt3/126m.yaml (hunk 2 above)
Comment (erhoo82) [reply launcher_scripts/conf/training/gpt3/126m.yaml:114]: Done

File: README.md (hunk 1 above)
Comment (erhoo82) [reply README.md]: Done

File: launcher_scripts/conf/training/gpt3/126m.yaml (hunk 2 above)
Comment (ericharper) [reply launcher_scripts/conf/training/gpt3/126m.yaml:114]: It's still True :D

Comment (ericharper) [reply launcher_scripts/conf/training/gpt3/126m.yaml]: N/m we will update the CI

File: launcher_scripts/conf/training/gpt3/126m.yaml [hunk 9]
```diff
@@ -109,14 +109,15 @@ model:
   grad_allreduce_chunk_size_mb: 125
 
   ## Transformer Engine
+  # fp8 training is currently not supported in the improved models
```
Comment (ericharper) [inline launcher_scripts/conf/training/gpt3/126m.yaml:112]: This is not an improved config. Let's leave it to True

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

File: launcher_scripts/conf/training/gpt3/126m.yaml (hunk 9 above)
Comment (erhoo82) [reply launcher_scripts/conf/training/gpt3/126m.yaml]: Done

**Pull request merged**